src/awkward/nplike.py

```diff
@@ -22,6 +22,10 @@ def of(*arrays):
                 nplikes.add(ak.nplike.Numpy.instance())
             elif type(array).__module__.startswith("cupy."):
                 nplikes.add(ak.nplike.Cupy.instance())
+            elif type(array).__module__.startswith("jaxlib"):
```

Comment:
Include a dot, `startswith("jaxlib.")`, so that it can't match a library named `jaxlibwhatever`.

Is it possible for relevant arrays to come from the `jax` library, rather than `jaxlib`? (I don't understand why JAX is split into two libraries like that, especially since the conda-forge recipe is broken: jax depends on a version of jaxlib that raises an exception on startup.)

If it's possible for the array to come from either `jaxlib` or `jax`, you'll need to check both `"jaxlib."` and `"jax."`.

Reply:
It's just `jaxlib.` since arrays come from XLA and `jaxlib.` is a wrapper around that. `jax` is a library that uses `jaxlib` so I think, just checking for `jaxlib.` will be sufficient for `JAX DeviceArrays`